File: Sort/InsertionSort/Ruby/insertion_sort_test.rb
require 'minitest/autorun'
require_relative 'insertion_sort'


describe "insertion_sort" do
  TEST_DATA = [8, 4, 3, 1, 2, 6, 5, 7];

  it "returns the position of the given value in the given list" do
    expect(insertion_sort(TEST_DATA)).must_equal([1, 2, 3, 4, 5, 6, 7, 8])
  end

  it "raises NoSuchValueError when the given value dose not exist in the list" do
    expect{ insertion_sort([]) }.must_raise(ArrayIsEmptyError)
  end
end

File: Sort/InsertionSort/TypeScript/insertionSort.test.ts
import { insertionSort } from './insertionSort'

describe('InsertionSort', () => {
  it('sorts the array in the right order', () => {
    const data = [8, 4, 3, 1, 2, 6, 5, 7]
    expect(insertionSort(data)).toStrictEqual([1, 2, 3, 4, 5, 6, 7, 8])
  })

  it('raises an error when the given array is empty', () => {
    expect(() => {
      insertionSort([])
    }).toThrow('The given array is empty')
  })
})
